synapse/handlers/event_auth.py

```diff
@@ -52,7 +52,7 @@ async def check_from_context(
             room_version_obj, event, auth_events=auth_events, do_sig_check=do_sig_check
         )
 
-    def compute_auth_events(
+    async def compute_auth_events(
```

Comment:
There's a few places that directly call `auth_types_for_event` instead of this:

* `FederationHandler._check_for_soft_fail`
* State resolution:
    * `v1._create_auth_events_from_maps`
    * `v1._resolve_auth_events`
    * `v2._iterative_auth_checks`

I haven't yet checked what (if any) changes need to be made to those.